webapp/src/js/services/telemetry.js

```diff
@@ -0,0 +1,215 @@
+var moment = require('moment'),
+  uuid = require('uuid/v4');
+
+angular
+  .module('inboxServices')
+  .factory('Telemetry', function($log, $q, $window, DB, Session) {
+    'use strict';
+    'ngInject';
+
+    var queue = $q.resolve();
+
+    // Intentionally scoped to the whole browser (for this domain). We can then
+    // tell if multiple users use the same device
+    var DEVICE_ID_KEY = 'medic-telemetry-device-id';
+
+    // Intentionally scoped to the specific user, as they may perform a
+    // different role (online vs. offline being being the most obvious) with
+    // different performance implications
+    var DB_ID_KEY = ['medic', Session.userCtx().name, 'telemetry-db'].join('-');
+    var LAST_AGGREGATED_DATE_KEY = [
+      'medic',
+      Session.userCtx().name,
+      'telemetry-date',
+    ].join('-');
```

Comment:
Rather than generating multiple user-specific keys for local storage and using these to persist primitive types, might it be simpler to have a single user-specific key and persist an object?  E.g.
```
localStorage.getItem(`medic-${userName}-telemetry-metadata`)
```
returning
```
{
  db_id: 'medic-user-myUsername-abc-123`,
  last_aggregated_date: 1234567890,
  some_key_added_in_the_future: 'value',
}
```
This question may be moot if a UUID is not required for the user's db (see below).

Reply:
I think local storage only stores strings ("DomStrings" even, however that is different), so it sort of nudges you into this pattern. I agree though, that packaging them up into one object would be nicer.

Reply:
Maybe you could `JSON.stringify()`, but perhaps then it's not really cleaning up the code :¬)